File: app/src/main/java/com/orangepenguin/boilerplate/util/FitCenterCropWhitePlaceholderFetcher.java
package com.orangepenguin.boilerplate.util;

import android.widget.ImageView;

import com.squareup.picasso.Picasso;

import javax.inject.Inject;

import static com.orangepenguin.boilerplate.BuildConfig.DEBUG;

/**
 * Fetches and location image and sets it on an {@link ImageView}.
 */
public class FitCenterCropWhitePlaceholderFetcher implements ImageFetcher<String, ImageView> {

    @Inject
    public FitCenterCropWhitePlaceholderFetcher() {
    }

    @Override
    public void fetchAndSetImage(String source, ImageView target) {
        Picasso picasso = Picasso.with(target.getContext());
        if (DEBUG) picasso.setIndicatorsEnabled(true);
        picasso.load(source)
                .placeholder(android.R.color.white)
                .fit()
                .centerCrop()
                .into(target);
    }
}

File: app/src/main/java/com/orangepenguin/boilerplate/util/FitCenterCropWhitePlaceholderFetcher.kt
package com.orangepenguin.boilerplate.util

import android.widget.ImageView
import com.orangepenguin.boilerplate.BuildConfig.DEBUG
import com.squareup.picasso.Picasso
import javax.inject.Inject

/**
 * Fetches and location image and sets it on an [ImageView].
 */
class FitCenterCropWhitePlaceholderFetcherKt @Inject
constructor() : ImageFetcher<String, ImageView> {

    override fun fetchAndSetImage(source: String, target: ImageView) {
        val picasso = Picasso.with(target.context)
        if (DEBUG) picasso.setIndicatorsEnabled(true)
        picasso.load(source)
                .placeholder(android.R.color.white)
                .fit()
                .centerCrop()
                .into(target)
    }
}
